src/components/Charts/RowChart/RowChart.js

```diff
@@ -0,0 +1,271 @@
+          tspan = innerText
+            .text(null)
+            .append('tspan')
+            .attr('x', spanWidth)
+            .attr('y', y)
+            .attr('dy', dy + 'em');
+
+        // eslint-disable-next-line no-cond-assign
+        while ((word = words.pop())) {
+          line.push(word);
+          tspan.text(line.join(' '));
+          if (tspan.node().getComputedTextLength() > width) {
+            line.pop();
+            tspan.text(line.join(' '));
+            line = [word];
+            tspan = innerText
+              .append('tspan')
+              .attr('x', spanWidth)
+              .attr('y', y)
+              // eslint-disable-next-line no-mixed-operators
+              .attr('dy', ++lineNumber * lineHeight + dy + 'em')
+              .text(word);
+            wrapCount++;
+          }
+        }
+
+        if (wrapCount) {
+          const viewMoreBackground = d3
+            .select(innerText.node().parentNode)
+            .select('.view-more-background');
+          const oldHeight = viewMoreBackground.attr('height');
+          // eslint-disable-next-line no-mixed-operators
+          const newHeight = parseFloat(oldHeight) + wrapCount * 12;
+          viewMoreBackground.attr('height', newHeight);
+        }
+      });
+      /* eslint-enable complexity */
+    };
+
+    const collapseARow = (rowName) => {
+      sendAnalyticsEvent('Bar chart collapsed', rowName);
+      dispatch(collapseRow(rowName));
+    };
+
+    const expandARow = (rowName) => {
+      sendAnalyticsEvent('Bar chart expanded', rowName);
+      dispatch(expandRow(rowName));
+    };
+
+    const selectFocus = (element) => {
+      const focusName = element.target.__data__;
+      // make sure to assign a valid lens when a row is clicked
+      const aLens = lens === 'Overview' ? 'Product' : lens;
+      const filters = coalesce(aggs, aLens.toLowerCase(), []);
+      scrollToFocus();
+      let values = [];
+      if (lens === 'Company') {
+        values.push(focusName.parent);
+      } else {
+        const filterGroup = filters.find((obj) => obj.key === focusName.parent);
+        const keyName = 'sub_' + lens.toLowerCase() + '.raw';
+        values = filterGroup
+          ? getAllFilters(focusName.parent, filterGroup[keyName].buckets)
+          : [];
+      }
+      sendAnalyticsEvent('Trends click', focusName.parent);
+      dispatch(changeFocus(focusName.parent, lens, [...values]));
+    };
+
+    const toggleRow = (element) => {
+      const rowName = element.target.__data__;
+      // fire off different action depending on if the row is expanded or not
+      const expandableRows = data
+        .filter((obj) => obj.isParent)
+        .map((obj) => obj.name);
+
+      if (!expandableRows.includes(rowName)) {
+        // early exit
+        return;
+      }
+
+      if (expandedRows.includes(rowName)) {
+        collapseARow(rowName);
+      } else {
+        expandARow(rowName);
+      }
+    };
+
+    const redrawChart = () => {
```

Comment:
does this still work without wrapping this function with 
`const redrawChart = () => {`

I'd like to make all of these chart components look as similar as possible so it's easier to make changes

Reply:
Yup it works without the function. Updated it as requested.